templates/epub3/toc.xhtml.ejs

```diff
@@ -11,9 +11,21 @@
 <nav id="toc" epub:type="toc">
     <ol>
         <% content.forEach(function(content, index){ %>
+            <% if(!content.excludeFromToc && content.beforeToc){ %>
+                <li class="table-of-content">
+                    <a href="<%= content.href %>"><%= (content.title || "Chapter "+ (1+index)) %><% if(content.author.length){ %> - <small class="toc-author"><%= content.author.join(",") %></small><% }%><% if(content.url){ %><span class="toc-link"><%= content.url %></span><% }%></a>
+                </li>
+            <% } %>
+        <% }) %>
         <li class="table-of-content">
-            <a href="<%= content.href %>"><%= (content.title || "Chapter "+ (1+index)) %><% if(content.author.length){ %> - <small class="toc-author"><%= content.author.join(",") %></small><% }%><% if(content.url){ %><span class="toc-link"><%= content.url %></span><% }%></a>
+            <a href="toc.xhtml">- <%= tocTitle %> -</a>
```

Comment:
@pedrosanta No, it's not `normal` to have a TOC content. but `beforeToc` and `afterToc` may make API more complicated.

How about make TOC an option content in the data array? like this:

```javascript
content: [{
        title: "About the author",
        author: "John Doe",
        data: "<h2>Charles Lutwidge Dodgson</h2>" +
            "<div lang=\"en\">Better known by the pen name Lewis...</div>"
    }, {
        title: "Copyright",
        data: "<h2>Copyright</h2>..."
    }, {
        type: "TOC" // => TOC here, will auto generate as normal content data.
    }, {
        title: "Down the Rabbit Hole",
        data: "<p>Alice was beginning to get very tired...</p>"
    }, {
        ...
    }
    ...
]
```

We can parse this data array, if found `data.type == 'TOC' `, replace it with a generated TOC html data. if not found `TOC` type, no TOC will insert into ebook, so someone can make their own TOC content, which can locate chapters with the `filename` attribute.

What do you think?  🙂